include/envoy/filesystem/filesystem.h

```diff
@@ -13,20 +14,29 @@
 namespace Envoy {
 namespace Filesystem {
 
+using FlagSet = std::bitset<4>;
+
 /**
  * Abstraction for a basic file on disk.
  */
 class File {
 public:
   virtual ~File() = default;
 
+  enum Operation {
+    Read,
+    Write,
+    Create,
+    Append,
+  };
+
   /**
-   * Open the file with O_RDWR | O_APPEND | O_CREAT
+   * Open the file with Flag
    * The file will be closed when this object is destructed
    *
    * @return bool whether the open succeeded
    */
-  virtual Api::IoCallBoolResult open() PURE;
+  virtual Api::IoCallBoolResult open(const FlagSet flags) PURE;
```

Comment:
I'm not sure of the actual implement of `bitset`. If we are passing by value remove the const, if we want to pass by reference, please do so.